apps/src/p5lab/spritelab/commands/criterionCommands.js

```diff
@@ -375,6 +375,60 @@ export const commands = {
     return false;
   },
 
+  // Returns true if a minimum number of sprites are in the specified group.
+  minimumGroupSprites(group, min = 1) {
+    const spriteIds = this.getSpriteIdsInUse();
+    const groupSpriteIds = spriteIds.filter(
+      id => this.nativeSpriteMap[id].group === group
+    );
+    return groupSpriteIds.length >= min;
+  },
+
+  // Returns true if a minimum number of sprites have no group.
+  minimumNonGroupSprites(min = 1) {
+    const spriteIds = this.getSpriteIdsInUse();
+    const groupSpriteIds = spriteIds.filter(
+      id => !this.nativeSpriteMap[id].group
+    );
+    return groupSpriteIds.length >= min;
+  },
+
+  // Returns true if sprites with a minimum number of costumes are in the specified group.
+  minimumCostumesForGroup(group, min = 1) {
+    const uniqueCostumes = [];
+    const spriteIds = this.getSpriteIdsInUse();
+    const spritesInUse = spriteIds.map(id => this.nativeSpriteMap[id]);
```

Comment:
Is `this.nativeSpriteMap[id]` an object that's either defined or undefined? (Just double-checking it's not a number)

Reply:
Yeah, looks like it is!

Reply:
Yeah, it should be a full sprite object not just an id. There are some mistakes to that effect in the commit history that I believe I fixed.